File: ranges_sorted_array/linear_2idx.py
class Solution:
    def summaryRanges(self, nums: List[int]) -> List[str]:
        res = []
        if len(nums) == 0:
            return []
        
        elif len(nums) == 1:
            return [str(nums[0])]

        j = 0
        n1 = nums[0]
        num_cnt = 0
        for i, num in enumerate(nums):
            if(j==0):
                j = 1
                continue

            if num == n1 + j:
                num_cnt += 1
                j += 1
                if not i == len(nums) - 1:
                    continue

            if num_cnt > 0:
                res.append(str(n1)+"->"+str(n1+num_cnt))
            elif num_cnt == 0:
                res.append(str(n1))

            if i == len(nums) - 1 and n1+num_cnt != num:
                res.append(str(num))

            n1 = num
            num_cnt = 0
            j = 1

        return res

File: ranges_sorted_array/linear_2idx.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    vector<string> summaryRanges(vector<int>& nums) {
        int idx1, idx2;
        vector<string> op_arr;
        // cout<<nums.size();
        int j = 0;
        if(nums.size() == 0)
            return op_arr;
        for(int i = 0; i < nums.size(); i++)
        {
            idx1 = nums[i];
            idx2 = nums[i+j];
            
            while ((idx2 - j) == idx1)
            {
                ++j;
                if(i+j >= nums.size())
                    break;
                idx2 = nums[i+j];
            }
            if(j==1)
            {
                op_arr.push_back(to_string(idx1));
            }
            else
            {
                op_arr.push_back(to_string(idx1) + "->" + to_string(nums[i+j-1]));
                i += j - 1;
            }
            j = 0;
            
        }
        return op_arr;
    }
};
